components/src/dynamo/vllm/handlers.py

```
@@ -625,6 +625,261 @@ async def wake_up(self, body: dict) -> dict:
+            async with lock:
+                lora_id = lora_name_to_id(lora_name)
+                is_hot_swap = lora_name in self.loaded_loras
+
+                # Hot-swap: vLLM's add_lora is a no-op when the lora_int_id is
+                # already registered, so we must remove the previous adapter
+                # first. remove_lora is best-effort on a fresh add.
+                if is_hot_swap:
+                    old_id = self.loaded_loras[lora_name].id
+                    try:
+                        await self.engine_client.remove_lora(old_id)
+                    except Exception as e:
```

Ruff (0.15.11)

[warning] Do not catch blind exception: `Exception`

(BLE001)